orderbook/src/main.rs

```diff
@@ -186,13 +184,11 @@ async fn main() {
         .expect("failed to create gas price estimator"),
     );
 
-    let pair_providers: Vec<Arc<dyn AmmPairProvider>> = stream::iter(args.shared.baseline_sources)
-        .then(|source| {
-            let web3 = web3.clone();
-            async move { sources::pair_provider(source, chain_id, &web3).await }
-        })
-        .collect()
-        .await;
+    let pair_providers = sources::pair_providers(&args.shared.baseline_sources, chain_id, &web3)
```

Comment:
It's pair providers still an appropriate name here (given that balancer is now incorporated as a liquidity source)?

Reply:
Balancer is skipped for pair providers (because as you mentioned it doesn't make sense for it). It is one of the APIs that are Uniswap specific that I want to tackle as tech debt this week:
> One thing that you may note from this PR is that some of the pool abstractions (pool aggregator, AMM pair provider, etc.) are quite specific to Uniswap-like liquidity. [...] I plan on tackling some of this tech debt in following PRs.